### PR TITLE
Add JSON autoindent regexes

### PR DESCRIPTION
**Pull request by Moosems**
JSON has a pretty simple syntax and, wanting a regex for my own editor, I decided to make it and then brought it over here :)

### CODE REVIEW

File: porcupine/default_filetypes.toml [hunk 1]
```diff
@@ -359,6 +359,7 @@ syntax_highlighter = "tree_sitter"
 tree_sitter_language_name = "json"
 indent_size = 2
 tabs2spaces = true
+autoindent_regexes = {dedent = '.*[^,{[]\n', indent = '.*[{[]\n'
```
Comment (rdbende) [inline porcupine/default_filetypes.toml:362]:
```suggestion
autoindent_regexes = {dedent = '.*[^,{[]\n', indent = '.*[{[]\n'}
```

Comment (Moosems) [reply porcupine/default_filetypes.toml:362]: I was doing this in the builtin GH editor so an issue was kinda to be expected.

**Comment by rdbende** [comment]
Needs tests (`tests/test_indent_dedent.py`), but otherwise it looks good to me.

**Comment by Moosems** [comment]
I found a few issues with the regexes so I'm fixing those right now.

**Comment by Moosems** [comment]
All normally formatted JSN files should be working.

**Comment by rdbende** [comment]
Does not unindent after end of list:
image

**Comment by Moosems** [comment]
:(. That's weird. I'll have to look into it more then, it worked when I tried it on a regex tester.

**Comment by rdbende** [comment]
Oh shit. It DOES work, it just forgot to git pull XD

**Comment by Moosems** [comment]
🤣. Lol!

**Comment by Moosems** [comment]
@rdbende Are you open today?

**Review by rdbende — APPROVED** [review]

**Comment by Moosems** [comment]
Yayyyyyyyy!

**Pull request merged**